### ISSUE TITLE
Bug in fmpz_poly_interpolate_fmpz_vec

### ISSUE THREAD

**Issue by thofma** [wbhart/flint2#362]
There is a bug in `fmpz_poly_interpolate_fmpz_vec`:
```julia
julia> R, x = PolynomialRing(ZZ, "x")
(Univariate Polynomial Ring in x over Integer Ring,x)

julia> f = interpolate(R, fmpz[1,2,3], fmpz[1, -4, 4])
6*x^2-23*x+18

julia> f(1)
1

julia> f(2)
-4

julia> f(3)
3
```

**Comment by thofma** [comment]
Ok, this is not a flint bug, since flint assumes that the polynomial has integer coefficients and returns garbage otherwise.

**Comment by wbhart** [comment]
The Flint function should raise an exception if it encounters an inexact division. So it's a valid bug, I think.

**Comment by wbhart** [comment]
Fixed in trunk.

**Comment by thofma** [comment]
Thanks.

**Comment by wbhart** [comment]
I'm hitting cases where Flint says the division is inexact, randomly from both the Nemo test code and from the Nemo benchmark code, even when the interpolation is guaranteed to exist. It should be relatively straightforward to generate examples, since I've seen it multiple times now.

**Comment by wbhart** [comment]
Actually, when I print out the input values, I get values which don't interpolate an integer polynomial. So something else is causing this bug. See the ticket here:

https://github.com/Nemocas/Nemo.jl/issues/231
